htmd/builder/amber.py

```diff
@@ -406,26 +431,40 @@ def build(mol, ff=None, topo=None, param=None, prefix='structure', outdir='./bui
 
     molbuilt = None
     if execute:
-        # Source paths of extra dirs (our dirs, not amber default)
-        htmdamberdir = os.path.abspath(os.path.join(home(), 'builder', 'amberfiles'))
-        sourcepaths = [htmdamberdir]
-        sourcepaths += [os.path.join(htmdamberdir, os.path.dirname(f))
-                        for f in ff if os.path.isfile(os.path.join(htmdamberdir, f))]
-        extrasource = []
-        for p in sourcepaths:
-            extrasource.append('-I')
-            extrasource.append('{}'.format(p))
+        if not teleapimports:
+            teleapimports = []
+            # Source default Amber (i.e. the same paths tleap imports)
+            amberhome = defaultAmberHome(teleap=teleap)
+            teleapimports.append(os.path.join(amberhome, 'dat', 'leap', 'prep'))
+            teleapimports.append(os.path.join(amberhome, 'dat', 'leap', 'lib'))
+            teleapimports.append(os.path.join(amberhome, 'dat', 'leap', 'parm'))
+            teleapimports.append(os.path.join(amberhome, 'dat', 'leap', 'cmd'))
+            if len(teleapimports) == 0:
+                raise RuntimeWarning('No default Amber force-field found. Check teLeap location: {}'.format(teleap))
+            # Source HTMD Amber paths that contain ffs
+            htmdamberdir = htmdAmberHome()
+            teleapimports += [os.path.join(htmdamberdir, os.path.dirname(f))
+                              for f in ff if os.path.isfile(os.path.join(htmdamberdir, f))]
+            if len(teleapimports) == 0:
+                raise RuntimeError('No default Amber force-field imports found. Check '
+                                   '`htmd.builder.amber.defaultAmberHome()` and `htmd.builder.amber.htmdAmberHome()`')
+        # Set import flags for teLeap
+        teleapimportflags = []
+        for p in teleapimports:
+            teleapimportflags.append('-I')
+            teleapimportflags.append('{}'.format(p))
```

Comment:
`'{}'.format(p)` seems excessive. Should be equivalent to `str(p)`

Reply:
sure, I can do that!